src/bin/leanc.in

```diff
@@ -1,49 +1,4 @@
 #!/usr/bin/env bash
-# Lean compiler
-#
-# A simple wrapper around a C compiler. Defaults to `@LEANC_CC@`,
-# which can be overridden with the environment variable `LEAN_CC`. All parameters are passed
-# as-is to the wrapped compiler.
-#
-# Interesting options:
-# * `-U LEAN_MULTI_THREAD` can be used to optimize programs not making use of multi-threading
-# * `--print-cflags`: print C compiler flags necessary for building against the Lean runtime and exit
-# * `--print-ldlags`: print C compiler flags necessary for statically linking against the Lean library and exit
-# * Set the `LEANC_GMP` environment variable to a path to `libgmp.a` (or `-l:libgmp.a` on Linux) to link GMP statically.
-#   Beware of the licensing consequences since GMP is LGPL.
-
-set -e
-bindir=$(dirname $0)
-
-cflags=("-I$bindir/../include" @LEANC_EXTRA_FLAGS@)
-ldflags=("-L$bindir/../lib/lean" "${LEANC_GMP:--lgmp}" @LEAN_EXTRA_LINKER_FLAGS@)
-ldflags_ext=(@LEANC_STATIC_LINKER_FLAGS@)
-args=("$@")
-for arg in "$@"; do
-    case $arg in
-        -shared)
-            # switch to shared linker flags
-            ldflags_ext=(@LEANC_SHARED_LINKER_FLAGS@)
-            ;;
-        -lleanshared | *libleanshared.*)
-            # linking against libleanshared explicitly (or linking libleanshared itself) ~> do not link against static stdlib
-            ldflags_ext=()
-            ;;
-        -c)
-            ldflags=()
-            ldflags_ext=()
-            ;;
-        --print-cflags)
-            echo "${cflags[@]}"
-            exit
-            ;;
-        --print-ldflags)
-            echo "${cflags[@]} ${ldflags_ext[@]} ${ldflags[@]}"
-            exit
-            ;;
-    esac
-done
-
-[ -n "$LEAN_CC" ] || LEAN_CC="@LEANC_CC@"
-
-$LEAN_CC "${cflags[@]}" "${args[@]}" "${ldflags_ext[@]}" "${ldflags[@]}" -Wno-unused-command-line-argument
+# used only for building Lean itself
+root=$(dirname $0)
+${LEAN_CC:-@CMAKE_C_COMPILER@} "-I$root/include" @LEANC_EXTRA_FLAGS@ "$@" "-L$root/lib/lean" "${LEANC_GMP:--lgmp}" @LEAN_EXTRA_LINKER_FLAGS@ -Wno-unused-command-line-argument
```

Comment:
do we really still need this leanc.sh then?  Can't we just move this line into the place that calls it? I think that's only stdlib.make.in right?

Reply:
We could, yes. It wouldn't exactly be pretty, and it makes it easier to communicate the flags to Nix.